src/plugins/charts/public/static/color_maps/color_maps.ts

```diff
@@ -54,3128 +62,86 @@ export interface ColorMap {
   [key: string]: RawColorSchema;
 }
 
+const COLOR_MAP_LENGTH = 512;
+
+function convertColorPaletteToColorMap(colorPalette: string[]): RawColorSchema['value'] {
```

Comment:
Can you add a TODO here with a link to an issue to refactor and simplify? As discussed offline, I think we can drastically reduce the length of this map, and we may also be able to refactor the map usage so that the explicitly calculated ranges and parsed color values are not necessary.

Otherwise this function would need a unit test, but I generally think the effort would be better spent removing the need for it.